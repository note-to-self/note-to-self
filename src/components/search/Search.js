import React from 'react';
import PropTypes from 'prop-types';
import styles from 'styled-components';

const FormStyles = styles.form`
@import url('https://fonts.googleapis.com/css?family=Muli:300,700');
@media (min-width: 375px) {
  font-family: 'Muli', sans-serif;
  font-size: 1.3em;
  text-align: left;
  padding-top: 2em;
}
`;

const Label = styles.label`
@media (min-width: 375px) {
  margin-bottom: 1em;
  padding-top: 1em;
  display: inline-block;
  text-align: left;
  vertical-align: 10px;
  margin-left: 1em;
  width: 50%
}
`;

const Input = styles.input`
@media (min-width: 375px) {
  margin-left: .2em;
  vertical-align: 10px;
  background: none;
  border-style: none;
  border-bottom: 1pt solid #568697;
  display: block;
  margin-right: auto;
  margin-left: auto;
  width: 80%;
  color: #568697;
}
`;

const Button = styles.button`
  display: block;
  margin-right: auto;
  margin-left: auto;
  width: 50%;
  margin-top: 1em;
  background-color: rgba(255, 255, 255, 0.9);
  border-style: none;
  padding: .5em;
  border-radius: 10px;
  font-family: 'Muli', sans-serif;
  color: #568697;
`;

export default function Search({ onChange, searchTerm }) {
  return (
    <>
      <FormStyles onSubmit={handleSubmit}>
        <Label>
          <Input type="text" name="searchTerm" value={searchTerm} onChange={onChange} ></Input>
        </Label>
        <Button>Search</Button>
      </FormStyles>
    </>
  );

}

Search.propTypes = {
  searchTerm: PropTypes.string,
  onChange: PropTypes.func,
};
