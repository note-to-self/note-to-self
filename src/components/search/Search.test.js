import React from 'react';
import Search from './Search';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

describe('Search', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Search />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
