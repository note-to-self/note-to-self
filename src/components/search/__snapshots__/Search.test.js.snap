// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches a snapshot 1`] = `
<form
  className="sc-bdVaJa dzKkSb"
>
  <label
    className="sc-bwzfXH cKJMRv"
  >
    <input
      className="sc-htpNat dbtHqZ"
      name="searchTerm"
      type="text"
    />
  </label>
  <button
    className="sc-bxivhb gqHkKN"
  >
    Search
  </button>
</form>
`;
